docs/datamodel/functions.rst

```diff
@@ -18,17 +18,39 @@ of :eql:type:`int64`:
     db> SELECT len({'hello', 'world'});
     {5, 5}
 
+This behavior is known as an *element-wise* operation. Many built-in
+and user-defined functions operate on elements. In case of multiple
+arguments, a cross-product of all the input sets is computed to
+determine all the input combinations that the function needs to be
+applied to. After that, the function is applied to each element of the
+cross-product in turn to generate the output set. This implies that if
```

Comment:
"in turn" -> "in order"? this sentence is a bit hard to read.

Reply:
This is hard to word right. The "in turn" here is referring to "the function is applied to each element". The order doesn't matter and it's just a way of saying "apply element-wise".

If I change it to "in order" it will look like it refers to "in order to generate the output set".

I think both of these are about the same level of awkward wording, so I've tried splitting the sentence instead.